lib/base-compiler.d.ts

```diff
@@ -3,10 +3,25 @@ export declare class BaseCompiler {
     public compiler;
     public lang;
     public outputFilebase: string;
+    protected mtime;
```

Comment:
These looks like general fixes, right? (Just for my understanding) - is there a way to have out linter somehow check this is in sync with base-compiler.js? (I don't fully undetstand why we have two files, but am following the pattern in my own code too, and this would seem to be a common issue)

Reply:
The `.d.ts` file exists solely to give type information about a JS file. The "only" way to have this synced would be to port base-compiler to ts althogether

Reply:
Of course! I was confused with the .interfaces.ts thing